src/Controllers/GetProductsController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Controllers;
+
+use App\Abstracts\Controller;
+use App\Interfaces\ProductModelInterface;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class GetProductsController extends Controller
+{
+    private $productModel;
+
+    /**
+     * GetProductsController constructor.
+     * @param $productModel
+     */
+    public function __construct(ProductModelInterface $productModel)
+    {
+        $this->productModel = $productModel;
+    }
+
+    public function __invoke(Request $request, Response $response, array $args)
+    {
+        $products = $this->productModel->getAllProducts();
+
+        if ($products) {
```

Comment:
we need separate responses depending on whether we couldn't connect to the database or whether there weren't any products in it

EDIT: maybe I'm misunderstanding the query return? based on my understanding:
it's false if the query doesn't work but returns the products as an array of Product Entities if the query does work
if the query works but there aren't any products in the DB, it returns an empty array, which is evaluated as false